### PR TITLE
New package: libvips-8.9.2

### PR DESCRIPTION
**Pull request by ifreund**
Also added `libmatio.so.9` to `commom/shlibs` as it was missing and
required for this package.

### CODE REVIEW

**Comment by ifreund** [comment]
So, all of the `makedepends` for libvips are optional and provide support for various image formats/other features.  I included everything for now as it seems like that's what most users would want but could make a default enabled build option for each optional dependency. Would this be desirable or be considered option bloat?

**Comment by Chocimier** [comment]
No need for build options, e.g. ImageMagick is similar.

Why only one arch?

Please rename template to vips, and install into it what is libvips_tools now.

**Comment by ifreund** [comment]
@Chocimier Made the changes you suggested, don't have a way to test the other archs myself but I guess we can rely on travis for that.

**Comment by Chocimier** [comment]
Separated binaries and library was good, please add libvips and rename subpackages to libvips-*.

Build option for hdf5 is needed, see `srcpkgs/armadillo/template`.

**Comment by ifreund** [comment]
Ah, I see what you mean now will do.

**Comment by ifreund** [comment]
Made the name changes and added the `hdf5` build option, but ran into an issue:

Are you sure that having `vips` be the template and `libvips` a subpackage is the right way to go? It seems to have messed up the `run_depends`/`shlib_requires` for the `vips` and `libvips` packages.

For `vips`:
```
run_depends:
	glib>=2.18.0_1
	glibc>=2.29_1
shlib-requires:
	libgobject-2.0.so.0
	libglib-2.0.so.0
	libc.so.6
	libpthread.so.0
```
For `libvips`:
```
run_depends:
	vips>=8.9.2_1
	libheif>=1.4.0_1
	...
shlib-provides:
	libvips-cpp.so.42
	libvips.so.42
shlib-requires:
	libvips.so.42
	libheif.so.1
	...
```

Which means that the produced `vips` package does not depend on `libvips` but fails to run unless `libvips` is installed. Maybe there's something I'm missing here, but adding `libvips` to `depends` failed to fix this.

**Comment by ifreund** [comment]
@Chocimier I believe I've fixed all the issues and the template now passes travis on all targets.  I went back to having `libvips` as the template, but with the subpackage containing the command line tools called`vips` instead of `libvips-tools` as I was unable to figure out a solution to the problem mentioned in the previous comment. Hopefully that is acceptable.

**Comment by Chocimier** [comment]
shlib-provides built correct with pkgname=vips and libvips in shlibs.

`case` block should be replaced with gir build helper, see `srcpkgs/atk/template`.

Please also indent contents of `makedepends=` by one space.

**Comment by ifreund** [comment]
Yeah, i think my environment must have been messy. I zapped everything and it worked fine with vips as the template as you said it would. Was not aware of the `gir` build helper, that's super handy!

Hopefully this is good to go now, I've certainly learned a lot about `xbps-src` working on it.

File: srcpkgs/vips/template [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+checksum=ae8491b1156cd2eb9cbbaa2fd6caa1dc9ed3ded0b70443d28cd7fea798ab2a27
+python_version=3
+
+build_options="gir hdf5"
+build_options_default="gir"
+desc_option_gir="GObject introspection"
+desc_option_hdf5="HDF5 support"
+
+if [ -z "$CROSS_BUILD" ]; then
+	build_options_default+="hdf5"
```
Comment (Chocimier) [inline srcpkgs/vips/template:45]: typo here, should have space in front when extending, please fix so we'll see build with options enabled on travis.

Comment (ifreund) [reply srcpkgs/vips/template]: Fixed!

**Comment by Chocimier** [comment]
Pushed with gir files moved to appropriate packages and removed gir option description, as this has default. Thanks!

**Comment by ifreund** [comment]
Wonderful, thanks!

**Pull request closed without merging**